op-supervisor/supervisor/backend/syncnode/controller.go

```diff
@@ -81,7 +86,7 @@ func (snc *SyncNodesController) AttachNodeController(chainID eth.ChainID, ctrl S
 	logger.Info("Attaching node", "chain", chainID, "passive", noSubscribe)
 
 	// create the managed node, register and return
-	node := NewManagedNode(logger, chainID, ctrl, snc.backend, noSubscribe)
+	node := NewManagedNode(logger, chainID, ctrl, snc.backend, snc.activationCheck, noSubscribe)
```

Comment:
Is this the reason for the `ActivationCheck` type? If yes, I think it's better to let the `ManagedMode` consume an interface that has the `Check...` function that it needs, and then just pass the dep set here.